site/documentation/content/api/telemetry-pubsub/index.md

```diff
@@ -0,0 +1,118 @@
+---
+title: "Telemetry API Specification for Google Pub/Sub"
+linkTitle: "Telemetry API (Pub/Sub)"
+weight: 405
+resources:
+- src: publish_pubsub_qos0.svg
+- src: publish_pubsub_qos1.svg
+---
+
+The *Telemetry* API is used by *Protocol Adapters* to send telemetry data downstream.
+*Business Applications* and other consumers use the API to receive data published by devices belonging to a particular
+tenant.
+
+The Telemetry API is defined by means of Google Pub/Sub message exchanges, i.e. a client needs to connect to Hono and configure
+Hono for Google Pub/Sub messaging infrastructure in order to use it and invoke operations of the API as described in the following sections. Throughout the remainder of
+this page we will simply use *Pub/Sub* when referring to Google Pub/Sub.
+
+The Telemetry API for Pub/Sub is an alternative to the [Telemetry API for Kafka]({{< relref "/api/telemetry-kafka" >}})
+for applications that want to consume telemetry data from a Pub/Sub Messaging Network instead of an Apache Kafka&trade;
+broker.
+
+## Southbound Operations
+
+The following operation can be used by *Protocol Adapters* to publish telemetry data received from devices to
+consumers like *Business Applications*.
+
+### Publish Telemetry Data
+
+The protocol adapter writes messages to the tenant-specific topic `projects/${google_project_id}/topics/${tenant_id}.telemetry`
+where `${google_project_id}` is the ID of the Google project and `${tenant_id}` is the
+ID of the tenant that the client wants to upload telemetry data for.
+
+**Preconditions**
+
+1. The Google Project ID is declared in the deployment file.
+1. The client is authorized to write to the topic.
+1. The device for which the adapter wants to send telemetry data has been registered (see
+   [Device Registration API]({{< relref "/api/device-registration" >}})).
+
+**Message Flow**
+
+It is up to the discretion of the protocol adapter whether it wants to use *AT LEAST ONCE* or *AT MOST ONCE* delivery
+semantics.
+
+The following sequence diagram illustrates the flow of messages involved in the *HTTP Adapter* publishing a telemetry
```

Comment:
while I believe that in theory, all adapters could use the south bound API, in reality you are implementing this only for the MQTT adapter, right? So I wonder if it would be more consistent (and less confusing) to use the MQTT adapter in the examples described below ...

Reply:
This should work for all adapters, since it is (like Kafka and AMQP) implemented in the abstract class of the protocol adapters. Currently we’ve tested it successfully with the HTTP and MQTT Adapter.
To keep it consistent with the other API documentation I have used the HTTP Adapter in the examples of the Telemetry API and the MQTT Adapter in the examples of the Event API.

Reply:
Does this mean that you will also implement this authentication scheme for the HTTP adapter? If so, then using the HTTP adapter would be fine with me. If not, then you should use the MQTT adapter in the example because otherwise readers might be surprised to find that the documentation describes behavior that is not working in practice.

Reply:
The Pub/Sub implementation is independent of the authentication scheme.
In the provided example everything is fine, because it is working with any existing authentication scheme and with different protocol adapters. It is just an additional messaging infrastructure.

Reply:
got it